Some Sundew compatibility is still broken for directories

https://github.com/MetPX/sarracenia/blob/9111a669094daf9da9050fa50d187b2254816d08/sarracenia/config.py#L2027-L2066


Test config (works for v2 or sr3):

```
broker amqps://dd.weather.gc.ca/
topicPrefix v02.post

subtopic bulletins.alphanumeric.*.*.CWAO.#

directory /tmp/sundew_matchPattern/${T1}${T2}/${CCCC}/${HH}
accept .*
```

v2 log:

```
2023-12-13 21:36:21,136 [INFO] file_log downloaded to: /tmp/sundew_matchPattern/UA/CWAO/21/UANT01_CWAO_132135___60774
2023-12-13 21:36:21,150 [INFO] file_log downloaded to: /tmp/sundew_matchPattern/UA/CWAO/21/UANT01_CWAO_132135___25215
2023-12-13 21:36:21,164 [INFO] file_log downloaded to: /tmp/sundew_matchPattern/IO/CWAO/21/IOBN05_CWAO_132120___19544
2023-12-13 21:36:21,177 [INFO] file_log downloaded to: /tmp/sundew_matchPattern/UA/CWAO/21/UANT01_CWAO_132135___16139
2023-12-13 21:36:21,190 [INFO] file_log downloaded to: /tmp/sundew_matchPattern/WS/CWAO/21/WSCN22_CWAO_132135___18982
2023-12-13 21:36:21,204 [INFO] file_log downloaded to: /tmp/sundew_matchPattern/WS/CWAO/21/WSCN02_CWAO_132135___39378
```

sr3 log:

```
2023-12-13 21:42:20,284 [INFO] 946630 sarracenia.flowcb.log after_work downloaded ok: /tmp/sundew_matchPattern/UA/CWAO/21/UANT01_CWAO_132139___23438
2023-12-13 21:42:20,284 [INFO] 946630 sarracenia.flowcb.log after_work downloaded ok: /tmp/sundew_matchPattern/CA/CWAO/21/CACN00_CWAO_132100__OWW_45242
2023-12-13 21:42:20,284 [INFO] 946630 sarracenia.flowcb.log after_work downloaded ok: /tmp/sundew_matchPattern/UA/CWAO/21/UANT01_CWAO_132139___52821
2023-12-13 21:42:20,284 [INFO] 946630 sarracenia.flowcb.log after_work downloaded ok: /tmp/sundew_matchPattern/UA/CWAO/21/UANT01_CWAO_132140___59862
2023-12-13 21:42:20,284 [INFO] 946630 sarracenia.flowcb.log after_work downloaded ok: /tmp/sundew_matchPattern/IO/CWAO/21/IOBN06_CWAO_132125___42394
2023-12-13 21:42:20,285 [INFO] 946630 sarracenia.flowcb.log after_work downloaded ok: /tmp/sundew_matchPattern/IS/CWAO/21/ISAA04_CWAO_132100_RRB__32877
2023-12-13 21:42:20,285 [INFO] 946630 sarracenia.flowcb.log after_work downloaded ok: /tmp/sundew_matchPattern/IS/CWAO/21/ISAB04_CWAO_132100_RRC__32409
2023-12-13 21:42:20,285 [INFO] 946630 sarracenia.flowcb.log after_work downloaded ok: /tmp/sundew_matchPattern/IS/CWAO/21/ISAB06_CWAO_132100_RRC__40994
```


But, mixing the sundew_matchPatterns things with sr3 % timestamps doesn't work right:

Converted sr3 config:

```
# created by: sr3 convert subscribe/test_sundew_matchPattern
broker amqps://dd.weather.gc.ca/
topicPrefix v02.post

subtopic bulletins.alphanumeric.*.*.CWAO.#

directory /tmp/sundew_matchPattern/${T1}${T2}/${CCCC}/${%H}
```

sr3 log

```
2023-12-13 21:39:28,286 [INFO] 946422 sarracenia.flow do_download missing destination directories, makedirs: %/tmp/sundew_matchPattern/{T1A/CWAO/21
2023-12-13 21:39:28,319 [INFO] 946422 sarracenia.flow do_download missing destination directories, makedirs: %/tmp/sundew_matchPattern/{T1A/CWAO/21
2023-12-13 21:39:28,349 [INFO] 946422 sarracenia.flow do_download missing destination directories, makedirs: %/tmp/sundew_matchPattern/{T1A/CWAO/21
2023-12-13 21:39:28,380 [INFO] 946422 sarracenia.flow do_download missing destination directories, makedirs: %/tmp/sundew_matchPattern/{T1A/CWAO/21
2023-12-13 21:39:28,418 [INFO] 946422 sarracenia.flowcb.log after_work downloaded ok: %/tmp/sundew_matchPattern/{T1A/CWAO/21/CACN00_CWAO_132100__WYW_00077
2023-12-13 21:39:28,418 [INFO] 946422 sarracenia.flowcb.log after_work downloaded ok: %/tmp/sundew_matchPattern/{T1A/CWAO/21/UANT01_CWAO_132138___11288
2023-12-13 21:39:28,418 [INFO] 946422 sarracenia.flowcb.log after_work downloaded ok: %/tmp/sundew_matchPattern/{T1A/CWAO/21/UANT01_CWAO_132138___51348
2023-12-13 21:39:28,418 [INFO] 946422 sarracenia.flowcb.log after_work downloaded ok: %/tmp/sundew_matchPattern/{T1A/CWAO/21/CACN00_CWAO_132100__OGP_11337
2023-12-13 21:39:28,418 [INFO] 946422 sarracenia.flowcb.log after_work downloaded ok: %/tmp/sundew_matchPattern/{T1A/CWAO/21/CACN00_CWAO_132100__WCI_27435
```

I put a bunch of log lines in:

```
2023-12-14 14:21:22,983 [ERROR] 1031665 sarracenia.config variableExpansion working on /tmp//20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/${%H}
2023-12-14 14:21:22,983 [ERROR] 1031665 sarracenia.config variableExpansion working on fragment /tmp//20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/
2023-12-14 14:21:22,983 [ERROR] 1031665 sarracenia.config variableExpansion close_brace > 0 /tmp//20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/
2023-12-14 14:21:22,983 [ERROR] 1031665 sarracenia.config variableExpansion working on fragment H}
2023-12-14 14:21:22,983 [ERROR] 1031665 sarracenia.config variableExpansion close_brace > 0 H}
2023-12-14 14:21:22,983 [ERROR] 1031665 sarracenia.config variableExpansion new dir changed to %/tmp//20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1${T2}/${CCCC}/14
2023-12-14 14:21:22,983 [ERROR] 1031665 sarracenia.config variableExpansion working on %/tmp//20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1${T2}/${CCCC}/14
2023-12-14 14:21:22,983 [ERROR] 1031665 sarracenia.config variableExpansion working on /tmp//20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/${%H}
2023-12-14 14:21:22,983 [ERROR] 1031665 sarracenia.config variableExpansion working on fragment /tmp//20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/
2023-12-14 14:21:22,983 [ERROR] 1031665 sarracenia.config variableExpansion close_brace > 0 /tmp//20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/
2023-12-14 14:21:22,983 [ERROR] 1031665 sarracenia.config variableExpansion working on fragment H}
2023-12-14 14:21:22,983 [ERROR] 1031665 sarracenia.config variableExpansion close_brace > 0 H}
2023-12-14 14:21:22,983 [ERROR] 1031665 sarracenia.config variableExpansion new dir changed to %/tmp//20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1${T2}/${CCCC}/14
2023-12-14 14:21:22,983 [ERROR] 1031665 sarracenia.config variableExpansion working on %/tmp//20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1${T2}/${CCCC}/14
2023-12-14 14:21:22,984 [ERROR] 1031665 sarracenia.config variableExpansion working on /tmp//20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/${%H}
2023-12-14 14:21:22,984 [ERROR] 1031665 sarracenia.config variableExpansion working on fragment /tmp//20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/
2023-12-14 14:21:22,984 [ERROR] 1031665 sarracenia.config variableExpansion close_brace > 0 /tmp//20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/
```

123c1f4 fixes the issue I was running into on ddsr dev.

Log before fix:

```
2023-12-14 17:21:09,992 [ERROR] sarracenia.config variableExpansion other fragments thing /apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/17
2023-12-14 17:21:09,992 [ERROR] sarracenia.config variableExpansion fragment is $/apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/
2023-12-14 17:21:09,992 [ERROR] sarracenia.config variableExpansion fragment is $T1}
2023-12-14 17:21:09,992 [ERROR] sarracenia.config variableExpansion fragment is $T2}/
2023-12-14 17:21:09,993 [ERROR] sarracenia.config variableExpansion fragment is $CCCC}/17
2023-12-14 17:21:09,993 [ERROR] sarracenia.config variableExpansion Modified new_dir is /apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/T1}T2}/CCCC}/17
```

Here's the log with the fix in place:

```
2023-12-14 17:22:39,435 [DEBUG] sarracenia.flow filter  urlToMatch: http://ddsr-cmc-dev01.cmc.ec.gc.ca/20231214/CCG/PRIVATE/SN/CWAO/17/SNVD02_CWAO_141700__VXKF_00553
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion fragments: ['/apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/17']
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion new_dir is now /apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/17
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion other fragments thing /apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/17
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion fragment is /apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion fragment is T1}
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion not a regex {fragment}
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion fragment is T2}/
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion not a regex {fragment}
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion fragment is CCCC}/17
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion not a regex {fragment}
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion Modified new_dir is /apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/17
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion fragments: ['/apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/17']
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion new_dir is now /apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/17
# note this is the start of regex substitution https://github.com/MetPX/sarracenia/blob/9111a669094daf9da9050fa50d187b2254816d08/sarracenia/config.py#L2231-L2257
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion other fragments thing /apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/17
2023-12-14 17:22:39,435 [ERROR] sarracenia.config variableExpansion fragment is /apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/
2023-12-14 17:22:39,436 [ERROR] sarracenia.config variableExpansion fragment is T1}
2023-12-14 17:22:39,436 [ERROR] sarracenia.config variableExpansion not a regex T1}
2023-12-14 17:22:39,436 [ERROR] sarracenia.config variableExpansion fragment is T2}/
2023-12-14 17:22:39,436 [ERROR] sarracenia.config variableExpansion not a regex T2}/
2023-12-14 17:22:39,436 [ERROR] sarracenia.config variableExpansion fragment is CCCC}/17
2023-12-14 17:22:39,436 [ERROR] sarracenia.config variableExpansion not a regex CCCC}/17
2023-12-14 17:22:39,436 [ERROR] sarracenia.config variableExpansion Modified new_dir is /apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/17
```

```python3
     assert try_pattern( options, message, "a_path/${DONT_TOUCH_THIS}something/else", r"a_path/\${DONT_TOUCH_THIS}something/else")
     assert try_pattern( options, message, "a_path/${1DONT_TOUCH_THIS}something/else", r"a_path/\${1DONT_TOUCH_THIS}something/else")
     assert try_pattern( options, message, "/apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/17", 
                        r"/apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/\${T1}\${T2}/\${CCCC}/17")
     assert try_pattern( options, message, "a_path/${1DONT_TOUCH_THIS}some/${%H}/thing/${STILL_DONT_CHANGE_ME}/${%Y}/hello", 
                        r"a_path/\${1DONT_TOUCH_THIS}some/[0-2][0-9]/thing/\${STILL_DONT_CHANGE_ME}/[0-9]{4}/hello")
```

these are the test cases i added in my branch. The last one is failing on the development branch:

```
>        assert try_pattern( options, message, "a_path/${1DONT_TOUCH_THIS}some/${%H}/thing/${STILL_DONT_CHANGE_ME}/${%Y}/hello",
                            r"a_path/\${1DONT_TOUCH_THIS}some/[0-2][0-9]/thing/\${STILL_DONT_CHANGE_ME}/[0-9]{4}/hello")
E        AssertionError: assert None
E         +  where None = try_pattern(<sarracenia.config.Config object at 0x7f43816f7910>, {'_format': 'v03', '_deleteOnPost': {'_format'}, 'source': 'WhereDataComesFrom', 'baseUrl': 'https://localhost/smorgasbord'}, 'a_path/${1DONT_TOUCH_THIS}some/${%H}/thing/${STILL_DONT_CHANGE_ME}/${%Y}/hello', 'a_path/\\${1DONT_TOUCH_THIS}some/[0-2][0-9]/thing/\\${STILL_DONT_CHANGE_ME}/[0-9]{4}/hello')

tests/sarracenia/config_test.py:58: AssertionError


DEBUG    sarracenia.config:config_test.py:27 input: a_path/${1DONT_TOUCH_THIS}some/${%H}/thing/${STILL_DONT_CHANGE_ME}/${%Y}/hello result: Thu_path/${1DONT_TOUCH_THISsome/18/thing/${STILL_DONT_CHANGE_ME}/2023/hello
```

--- a/sarracenia/config.py
+++ b/sarracenia/config.py
@@ -2150,9 +2150,10 @@ class Config:
 
 
         # strftime compatible patterns.
-        if len(new_dir.split( '${%' )) > 1:
-            fragment_list=[]
-            for fragment in new_dir.split( '${%' ):
+        fragments = new_dir.split( '${%' )
+        if len(fragments) > 1:
+            fragment_list=[fragments[0]]
+            for fragment in fragments[1:]:
                 close_brace = fragment.find('}')
                 frag_start=0
                 seconds=self.varTimeOffset

--- a/sarracenia/config.py
+++ b/sarracenia/config.py
@@ -2243,7 +2243,7 @@ class Config:
                 matches= re.search( r'^[0-9]+$', match_field)
                 # non-numeric thing... variable or something.
                 if not matches:
-                    fragment_list.append(fragment)
+                    fragment_list.append('${' + fragment)
                     continue
                 field=int(match_field)
                 if self.sundew_compat_regex_first_match_is_zero:

--- a/tests/sarracenia/config_test.py
+++ b/tests/sarracenia/config_test.py
@@ -51,10 +51,12 @@ def test_variableExpansion():
             "/pub/DATASETS/NOAA/G02158/unmasked" + '/[0-9]{4}/[0-9]{2}_[A-Z]{1}[a-z]{2}' )
      assert try_pattern(  options, message, "${PBD}/${%Y}/${SOURCE}/AIRNOW/CSV/BCMOE/${%H}/",  \
              options.post_baseDir + '/[0-9]{4}/'+ message['source'] +'/AIRNOW/CSV/BCMOE/[0-9]{2}/' )
-     assert try_pattern( options, message, "a_path/${DONT_TOUCH_THIS}something/else", "a_path/\${DONT_TOUCH_THIS}something/else")
-     assert try_pattern( options, message, "a_path/${1DONT_TOUCH_THIS}something/else", "a_path/\${1DONT_TOUCH_THIS}something/else")
+     assert try_pattern( options, message, "a_path/${DONT_TOUCH_THIS}something/else", r"a_path/\${DONT_TOUCH_THIS}something/else")
+     assert try_pattern( options, message, "a_path/${1DONT_TOUCH_THIS}something/else", r"a_path/\${1DONT_TOUCH_THIS}something/else")
      assert try_pattern( options, message, "/apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/17", 
-                        "/apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/\${T1}\${T2}/\${CCCC}/17")
+                        r"/apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/\${T1}\${T2}/\${CCCC}/17")
+     assert try_pattern( options, message, "a_path/${1DONT_TOUCH_THIS}some/${%H}/thing/${STILL_DONT_CHANGE_ME}/${%Y}/hello", 
+                        r"a_path/\${1DONT_TOUCH_THIS}some/[0-2][0-9]/thing/\${STILL_DONT_CHANGE_ME}/[0-9]{4}/hello")
 
      options.sundew_compat_regex_first_match_is_zero = True
      options.post_baseDir = '/apps/sarra/public_data'

--- a/tests/sarracenia/config_test.py
+++ b/tests/sarracenia/config_test.py
@@ -51,6 +51,10 @@ def test_variableExpansion():
             "/pub/DATASETS/NOAA/G02158/unmasked" + '/[0-9]{4}/[0-9]{2}_[A-Z]{1}[a-z]{2}' )
      assert try_pattern(  options, message, "${PBD}/${%Y}/${SOURCE}/AIRNOW/CSV/BCMOE/${%H}/",  \
              options.post_baseDir + '/[0-9]{4}/'+ message['source'] +'/AIRNOW/CSV/BCMOE/[0-9]{2}/' )
+     assert try_pattern( options, message, "a_path/${DONT_TOUCH_THIS}something/else", "a_path/\${DONT_TOUCH_THIS}something/else")
+     assert try_pattern( options, message, "a_path/${1DONT_TOUCH_THIS}something/else", "a_path/\${1DONT_TOUCH_THIS}something/else")
+     assert try_pattern( options, message, "/apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/${T1}${T2}/${CCCC}/17", 
+                        "/apps/sarra/public_data/20231214/SSC-DATAINTERCHANGE/MSC-BULLETINS/\${T1}\${T2}/\${CCCC}/17")
 
      options.sundew_compat_regex_first_match_is_zero = True
      options.post_baseDir = '/apps/sarra/public_data'
@@ -87,4 +91,3 @@ def test_variableExpansion():
 
      # to get stuff to print out, make it fail.
      #assert False 
-     

